packages/ovm/src/contracts/ExecutionManager.sol

```diff
@@ -557,10 +571,17 @@ contract ExecutionManager is FullStateManager {
      *       [methodID (bytes4)]
      *       [salt (bytes32)]
      *       [ovmInitcode (bytes (variable length))]
-     * returndata: [newOvmContractAddress (as bytes32)]
+     * returndata: [newOvmContractAddress (as bytes32)] -- will be all 0s if this create failed.
      */
     function ovmCREATE2() public {
-        require(!executionContext.inStaticContext, "Cannot create new contracts from a STATICCALL.");
+        if (executionContext.inStaticContext) {
+            // Cannot create new contracts from a STATICCALL -- return 0 address
+            assembly {
+                let returnData := mload(0x40)
+                mstore(returnData, 0)
+                return(returnData, 0x20)
+            }
+        }
```

Comment:
Do we know that this is what is returned during a failed STATIC_CALL? I personally don't so just want to make sure

Reply:
Yep! https://github.com/ethereum-optimism/optimism-monorepo/pull/10/files#diff-712fcb88c95eed4ca84335dd59b7c435R440

Reply:
I suppose we had tests for all failures result in 0 address and that static calls to create fail, but I added some tests to make sure that static calls actually return 0 address 👍